offline/coturn.md

```diff
@@ -0,0 +1,471 @@
+define ASSETHOSTIP= 192.168.122.10
+
+define INF_WAN    = enp41s0
+``` 
+
+Where:
+
+* `COTURNIP` is the IP address of the machine where Coturn will run (in our example, the first kubernetes node, `kubenode1`).
+* `SFTIP` is the IP address of the machine where SFT will run (in our example, the third kubernetes node, `kubenode3`).
+* `ANSNODEIP` is the IP address the first machine where ansible will install non-kubernetes services (in our example, the first ansible node, `ansnode1`).
+* `ASSETHOSTIP` is the IP address of the machine where the assethost will run (see earlier steps in the installation process.)
+* `INF_WAN` is the name of the WAN interface exposed to the outside world (the Internet).
+
+Then, we edit the `table ip nat` / `chain PREROUTING` section of the file:
+
+```nft
+
+table ip nat {
+  chain PREROUTING {
+
+    type nat hook prerouting priority -100;
+
+    iifname { $INF_WAN, virbr0 } tcp dport 80  fib daddr type local dnat to $SFTIP:31772
+    iifname { $INF_WAN, virbr0 } tcp dport 443 fib daddr type local dnat to $SFTIP:31773
+
+    udp dport 80   dnat ip to $ANSNODEIP:80
+    udp dport 1194 dnat ip to $ASSETHOSTIP:1194
+
+    iifname $INF_WAN ip daddr 5.9.84.121 udp dport 32768-46883 dnat to $COTURNIP
+    iifname $INF_WAN ip daddr 5.9.84.121 udp dport 46884-61000 dnat to $SFTIP
+
+    iifname $INF_WAN udp dport 3478 dnat to $COTURNIP:3478
+    iifname $INF_WAN tcp dport 3478 dnat to $COTURNIP:3478
+
+    fib daddr type local counter jump DOCKER
+  }
+
+``` 
+
+Some explanations:
+
+This is used for the SFT control:
+
+```nft 
+    iifname { $INF_WAN, virbr0 } tcp dport 80  fib daddr type local dnat to $SFTIP:31772
+    iifname { $INF_WAN, virbr0 } tcp dport 443 fib daddr type local dnat to $SFTIP:31773
+``` 
+
+This is the part that distributes the UDP packets (media/calling traffic) in two different port ranges for SFT and Coturn:
+
+```nft
+    iifname $INF_WAN ip daddr 5.9.84.121 udp dport 32768-46883 dnat to $COTURNIP
+    iifname $INF_WAN ip daddr 5.9.84.121 udp dport 46884-61000 dnat to $SFTIP
+``` 
+
+This is the part that redirects the control traffic to the Coturn port:
+
+```nft
+    iifname $INF_WAN udp dport 3478 dnat to $COTURNIP:3478
+    iifname $INF_WAN tcp dport 3478 dnat to $COTURNIP:3478
+```
+
+
+Then we restart Nftables to apply the changes:
+
+```bash
+
+sudo systemctl restart nftables
+
+```
+
+## Change the Wire-Server configuration to use Coturn instead of Restund.
+
+We must change the Wire-Server configuration to use Coturn instead of Restund.
+
+First, we must locate what the "external" IP address of the machine is. 
+
+This is the IP we must provide in our Wire-Server configuration to allow the clients to connect to Coturn. 
+
+We get it by running the following command:
+
+```bash
+
+sudo ip addr
+
+```
+
+The first interface will be the loopback interface, `lo`, and the second interface will be the "external" interface, `enp41s0` in our example, the output looking something like this:
```

Comment:
What is the intended audience of this guide? Shouldn't we assume that someone following the guide is familiar with IP addresses and can figure out the IP address of a machine, i.e. we don't need to spell `sudo ip addr` for them?

Reply:
@marcoconti83 the general guidance for docs is we should be on the side of explaining more rather than explaining less. most users are tech-savvy, but I've had to walk some through some very basic things (even more basic than this). I asked Julia and she confirmed in this case it's fine.